dev/download-github-artefact

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
```

Comment:
Perhaps this script filename should end with .py and it should be invoked by an explicit `python` call, so that we control which Python executable runs it?

Reply:
> Do I understand correctly that this PR does all of the following?

Yes, although we also update the theme to the new pydata theme. We don't yet take full advantage of this / make any sweeping design changes; that's down the road.

> If I merge this PR, what happens to the currently live documentation? 

Any activity on a PR will update that PR branch on RTD. The per-branch visibility is controlled on RTD side. However, I haven't yet set up the building from `main` - I wanted to run the rest of this PR past you first! It might make sense to trigger this workflow on `branch` activity rather than PRs.

All doc-building is configured from GitHub via our CI and the `.readthedocs.yaml` configuration. 

> Would it be appropriate for me to redirect https://awkward-array.org to point to https://awkward-array.readthedocs.io so that they become synonyms, and I can cancel Netlify?

Yes - and I'd suggest that we make use of custom domains to have `awkward-array` point to the RTD: https://docs.readthedocs.io/en/stable/custom-domains.html

> When would new documentation go to readthedocs? With every commit, with every PR, or with every release/tag? Ideally, it should be tested in every commit and deployed to readthedocs with every release/tag.

Currently with every "event" on a PR. This should probably change to a per-branch workflow which is more intuitive (and will then naturally pick-up merges). TODO...

 > Can I see the new documentation before committing? Is it on an unlisted readthedocs URL?
 
 The docs are live, but unlisted: https://awkward-array.readthedocs.io/en/feat-merge-docs/index.html

Reply:
> Perhaps this script filename should end with .py and it should be invoked by an explicit python call, so that we control which Python executable runs it?

Sure! I have a preference for executables, but it's really no big deal either way.

Reply:
NB: we'll have 50% failing builds until we remove the built-in webhook from RTD, as it does not wait for the artefacts to be deployed by the docs workflow